src/Interpreters/SystemLog.h

```diff
@@ -229,9 +230,18 @@ void SystemLog<LogElement>::startup()
 }
 
 
+static thread_local bool recursive_add_call = false;
+
 template <typename LogElement>
 void SystemLog<LogElement>::add(const LogElement & element)
 {
+    /// It is possible that the method will be called recursively.
+    /// Better to drop these events to avoid complications.
+    if (recursive_add_call)
+        return;
+    recursive_add_call = true;
+    SCOPE_EXIT({ recursive_add_call = false; });
```

Comment:
Why not simply `MemoryTracker::BlockerInThread temporarily_disable_memory_tracker(VariableContext::Global)` ?

Reply:
Current solution is more generic - it will avoid all possible recursive function calls.
Also total memory will be accounted correctly.